### DISCUSSION TITLE
getContract type issue with ts strict mode off after 1.15.2

### DISCUSSION THREAD

**Discussion by imaksp** [wevm/viem#1304] [General]
I know that ts strict mode is required for this lib, but adding for info that this [commit](https://github.com/wagmi-dev/viem/commit/de24c066e905448f778b0c30668f3bf279b3fe7b) broke the types with strict false, before 1.15.2 it was working, ibut now it is giving this error when walletClient is passed in getContract:

> The expected type comes from property 'walletClient' which is declared here on type 'GetContractParameters<Transport, Chain, Account, readonly [{ readonly type: "event"; readonly name: "Approval"; readonly inputs: readonly [{ readonly indexed: true; readonly name: "owner"; readonly type: "address"; }, { ...; }, { ...; }]; }, ... 9 more ..., { ...; }], never, never, `0x${string}`>'

    node_modules/viem/_types/actions/getContract.d.ts:42:4:
      42 │     walletClient?: TWalletClient;

Let me know if there is any way to make it work with strict false or by adding some extra option in tsconfig.

**Comment by tmm** [comment]
If this is a bug, can you open an issue with a reproduction (TypeScript Playground is fine)?
